src/cluster/clusterServices.tsx

```diff
@@ -0,0 +1,360 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import {
+  API_HEADER_BEARER,
+  API_HEADER_CONTENT_TYPE,
+  ClusterStatus,
+  HTTP_METHOD,
+  POLLING_TIME_LIMIT,
+  gcpServiceUrls
+} from '../utils/const';
+import {
+  authApi,
+  toastifyCustomStyle,
+  loggedFetch,
+  getProjectId,
+  authenticatedFetch,
+  statusValue
+} from '../utils/utils';
+import { DataprocLoggingService, LOG_LEVEL } from '../utils/loggingService';
+
+interface IClusterRenderData {
+  status: { state: ClusterStatus };
+  clusterName: string;
+}
+
+interface IClusterDetailsResponse {
+  error: {
+    message: string;
+    code: number;
+  };
+  status: {
+    state: string;
+  };
+  clusterName: string;
+  clusterUuid: string;
+  projectId?: string;
+  regionId?: string;
+}
+export class ClusterService {
+  static listClustersAPIService = async (
+    setProjectId: (value: string) => void,
+    renderActions: (value: IClusterRenderData) => React.JSX.Element,
+    setclustersList: (value: any) => void,
```

Comment:
type please!

Reply:
also setClustersList

Reply:
Resolved: type added and setClustersList changed